components/service/glean/src/test/java/mozilla/components/service/glean/storages/StringListsStorageEngineTest.kt

```diff
@@ -0,0 +1,258 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean.storages
+
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.test.core.app.ApplicationProvider
+import mozilla.components.service.glean.Lifetime
+import mozilla.components.service.glean.StringListMetricType
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.ArgumentMatchers
+import org.mockito.Mockito
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class StringListsStorageEngineTest {
+
+    @Before
+    fun setUp() {
+        StringListsStorageEngine.applicationContext = ApplicationProvider.getApplicationContext()
+        // Clear the stored "user" preferences between tests.
+        ApplicationProvider.getApplicationContext<Context>()
+            .getSharedPreferences(
+                StringListsStorageEngine.javaClass.simpleName,
+                Context.MODE_PRIVATE)
+            .edit()
+            .clear()
+            .apply()
+        StringListsStorageEngine.clearAllStores()
+    }
+
+    @Test
+    fun `set() properly sets the value in all stores`() {
+        val storeNames = listOf("store1", "store2")
+
+        val metric = StringListMetricType(
+            disabled = false,
+            category = "telemetry",
+            lifetime = Lifetime.Ping,
+            name = "string_list_metric",
+            sendInPings = storeNames
+        )
+
+        val list: MutableList<String> = mutableListOf()
```

Comment:
nit: you can collapse lines 49, 51 and 52 as `val list = listOf("First", "Second")`